design_notes/XDN11 - Error handling.md

```diff
@@ -0,0 +1,345 @@
+---
+id: XDN11
+title: Error handling
+author: mawign@microsoft.com
+status: Draft
+---
+
+# XDN11 - Error handling
+
+- Author: Manodasan Wignarajah (mawign@microsoft.com)
+- Status: Draft
+
+## Abstract
+
+This document describes the error handling story for xlang and what it will support.
+
+## Overview
+
+As described in XDN01, xlang is a system that allows a component written in any supported language
+to be callable from any other supported language. To achieve this, xlang needs to support a way to
+translate errors from one supported language to another supported language.
+
+These translated errors may end up being handled or end up as application failures which are
+debugged by a developer, logged by the application, or captured in a dump. To help facilitate
```

Comment:
Is it worth calling at high level the difference in expectations for "implementation debugging information" and "external facing pretty errors"? When designing similar copmponents for Project Rome we took a stance similar to WinRT's RoOriginateError which is intended for an audience external to the component vs exception messages and general logging that are usually for the benefit of the component developer. These lines obviously get blurred as a consumer reports issues to the developer but we felt it was an important audience distinction to call out and make sure a solution addressed the needs of both parties.

Reply:
As per arch meeting, @bbowman further expounded on this topic. In particular, he clarified the idea of different audiences for error info - the component owner vs. the component user (aka developer) vs the app user (non developer). Makes sense to call this out explicitly. Also, maybe worth calling out how component user can communicate to the component owner to distinguish between "user is using the component wrong" vs. "there's a bug in the component". Having this called out in the overview seems reasonable as we describe further parts of the error model later in the doc

Reply:
Called this out in the overview.